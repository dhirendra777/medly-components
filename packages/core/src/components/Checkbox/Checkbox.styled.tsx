import { CheckIcon } from '@medly-components/icons';
import { defaultTheme } from '@medly-components/theme';
import { centerAligned, styled, WithThemeProp } from '@medly-components/utils';
import { CheckboxWrapperProps, Props } from './types';

export const Wrapper = styled('div')<CheckboxWrapperProps>`
    position: relative;
    width: ${({ theme, size }) => (size ? theme.checkbox.sizes[size] : theme.checkbox.defaultSize)};
    height: ${({ theme, size }) => (size ? theme.checkbox.sizes[size] : theme.checkbox.defaultSize)};
    border: 1px solid ${({ theme }) => theme.checkbox.borderColor};
    border-radius: 15%;

    &:focus-within {
        box-shadow: 0 0 0 1pt ${({ theme }) => theme.checkbox.outlineColor};
    }

    & > * {
        width: 100%;
        height: 100%;
        border-radius: 15%;
    }

    ${centerAligned()}
`;
Wrapper.defaultProps = {
    theme: defaultTheme
};

export const CheckIconWrapper = styled('div')<WithThemeProp>`
    z-index: 1;
    position: absolute;
    background-color: ${({ theme }) => theme.checkbox.bgColor};

    ${CheckIcon.Style} {
        width: 0;
        height: 0;
        transition: width 0.2s ease, height 0.2s ease;
    }

    ${centerAligned()}
`;
CheckIconWrapper.defaultProps = {
    theme: defaultTheme
};

export const Checkbox = styled('input').attrs({ type: 'checkbox' })<Props>`
    opacity: 0;
    z-index: 2;
    position: absolute;
    margin: 0;
    cursor: pointer;

    &:focus {
        outline: none;
    }

    &:disabled {
        cursor: not-allowed;
        & + ${CheckIconWrapper} {
            background: ${({ theme }) => theme.checkbox.disabledBgColor};
        }
    }

    &:checked + ${CheckIconWrapper} {
        ${CheckIcon.Style} {
            width: calc(70%);
            height: calc(70%);
        }
    }
`;
