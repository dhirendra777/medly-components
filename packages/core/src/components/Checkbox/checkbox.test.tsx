import { fireEvent, render } from '@test-utils';
import React from 'react';
import { Checkbox } from './Checkbox';

describe('Checkbox component', () => {
    it('should render correctly with all the default props', () => {
        const { container } = render(<Checkbox name="gender" />);
        expect(container).toMatchSnapshot();
    });

    it('should render correctly with all the props given', () => {
        const { container } = render(<Checkbox disabled fullWidth name="gender" label="Female" size="M" labelPosition="right" />);
        expect(container).toMatchSnapshot();
    });

    it('should call onChange handler on click on checkbox', () => {
        const mockOnChange = jest.fn();
        const { container } = render(
            <Checkbox onChange={mockOnChange} disabled name="gender" label="Female" size="M" labelPosition="right" />
        );
        fireEvent.click(container.querySelector('input'));
        expect(mockOnChange).toBeCalled();
    });
});
