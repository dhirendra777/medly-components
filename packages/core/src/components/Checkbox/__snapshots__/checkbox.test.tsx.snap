// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should render correctly with all the default props 1`] = `
.c6 {
  font-size: 16px;
  cursor: pointer;
}

.c6 * {
  fill: #616161;
}

.c0 {
  margin: 8px 8px 8px 0px;
  display: inline-grid;
  grid-template-columns: max-content minmax(max-content,25%);
  grid-template-areas: 'field label' 'description random';
  grid-template-columns: auto;
  grid-template-areas: 'field' 'description';
}

.c0 .c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c1 {
  position: relative;
  width: 16px;
  height: 16px;
  border: 1px solid #E0E0E0;
  border-radius: 15%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:focus-within {
  box-shadow: 0 0 0 1pt #012040;
}

.c1 > * {
  width: 100%;
  height: 100%;
  border-radius: 15%;
}

.c4 {
  z-index: 1;
  position: absolute;
  background-color: #ffffff;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 .c5 {
  width: 0;
  height: 0;
  -webkit-transition: width 0.2s ease,height 0.2s ease;
  transition: width 0.2s ease,height 0.2s ease;
}

.c2 {
  opacity: 0;
  z-index: 2;
  position: absolute;
  margin: 0;
  cursor: pointer;
}

.c2:focus {
  outline: none;
}

.c2:disabled {
  cursor: not-allowed;
}

.c2:disabled + .c3 {
  background: #E0E0E0;
}

.c2:checked + .c3 .c5 {
  width: calc(70%);
  height: calc(70%);
}

<div>
  <div
    class="c0"
    id="-checkbox"
  >
    
    <div
      class="c1"
    >
      <input
        class="c2"
        id=""
        name="gender"
        type="checkbox"
      />
      <div
        class="c3 c4"
      >
        <svg
          class="c5 c6"
          height="1em"
          viewBox="0 0 18 14"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 11.17L1.83 7 .41 8.41 6 14 18 2 16.59.59z"
            fill="none"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Checkbox component should render correctly with all the props given 1`] = `
.c8 {
  font-size: 16px;
  cursor: pointer;
}

.c8 * {
  fill: #616161;
}

.c1 {
  font-size: 14px;
  font-weight: 400;
  color: #616161;
  margin: 0px 0px 0px 8px;
  cursor: not-allowed;
}

.c2 {
  grid-area: label;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  margin: 8px 0px;
  display: grid;
  grid-template-columns: max-content minmax(max-content,25%);
  grid-template-areas: 'field label' 'description random';
}

.c0 .c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c3 {
  position: relative;
  width: 24px;
  height: 24px;
  border: 1px solid #E0E0E0;
  border-radius: 15%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3:focus-within {
  box-shadow: 0 0 0 1pt #012040;
}

.c3 > * {
  width: 100%;
  height: 100%;
  border-radius: 15%;
}

.c6 {
  z-index: 1;
  position: absolute;
  background-color: #ffffff;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 .c7 {
  width: 0;
  height: 0;
  -webkit-transition: width 0.2s ease,height 0.2s ease;
  transition: width 0.2s ease,height 0.2s ease;
}

.c4 {
  opacity: 0;
  z-index: 2;
  position: absolute;
  margin: 0;
  cursor: pointer;
}

.c4:focus {
  outline: none;
}

.c4:disabled {
  cursor: not-allowed;
}

.c4:disabled + .c5 {
  background: #E0E0E0;
}

.c4:checked + .c5 .c7 {
  width: calc(70%);
  height: calc(70%);
}

<div>
  <div
    class="c0"
    id="Female-checkbox"
  >
    <label
      class="c1 c2"
      for="Female"
    >
      Female
    </label>
    <div
      class="c3"
      disabled=""
    >
      <input
        class="c4"
        disabled=""
        id="Female"
        name="gender"
        type="checkbox"
      />
      <div
        class="c5 c6"
      >
        <svg
          class="c7 c8"
          height="1em"
          viewBox="0 0 18 14"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 11.17L1.83 7 .41 8.41 6 14 18 2 16.59.59z"
            fill="none"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
